extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -155,8 +155,15 @@ export class WorkspaceService implements IWorkspaceService {
 		return projectTypes;
 	}
 
-	getProjectsInWorkspace(): vscode.Uri[] {
-		return vscode.workspace.workspaceFile ? this.getWorkspaceConfigurationValue<string[]>(ProjectsConfigurationName).map(project => this.toUri(project)) : [];
+	getProjectsInWorkspace(ext?: string): vscode.Uri[] {
+		let projects = vscode.workspace.workspaceFile ? this.getWorkspaceConfigurationValue<string[]>(ProjectsConfigurationName).map(project => this.toUri(project)) : [];
+
+		// filter by specified extension
+		if (ext) {
+			projects.filter(p => path.extname(p.fsPath) === ext);
```

Comment:
1. we should handle the casing as well as whether there is a leading '.'. `sqlProj`, and `.sqlproj` should both work
2. you have to assign this to the projects variable, filter method won't update the original array

Reply:
oops thanks for catching